### PR TITLE
 kde-applications: 18.12.1 -> 18.12.3

### PR DESCRIPTION
**Pull request by nyanloutre**
###### Motivation for this change

Also including improvement for the `fetch-kde-qt.sh` script. I changed it so it only needs to download metadata from the KDE and Qt website instead of every source tarball. Because of this the `fetch.sh` needs to be updated and may cause conflicts with already opened PR like #55847 and #55650 (I could update mt branch accordingly once they are merged)

I also fixed a conflict with `akonadi-paths.patch` patch.

Also should I update existing hashes in `src.nix` ? This would only convert existing base32 hashes into hexa.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS (only some packages)
   - [ ] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
   - `nixos.tests.plasma5.x86_64-linux`
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by dtzWill** [comment]
https://github.com/NixOS/nixpkgs/pull/55549

But this looks better, so assuming this is the same + fix I can close the other :).

**Comment by nyanloutre** [comment]
> #55549
> 
> But this looks better, so assuming this is the same + fix I can close the other :).

Oh sorry, I didn't saw your PR before opening this one

File: maintainers/scripts/fetch-kde-qt.sh [hunk 1]
```diff
@@ -38,8 +39,8 @@ gawk -F , "{ print \$1 }" $csv | sort | uniq | while read name; do
     latestVersion=$(echo "$versions" | sort -rV | head -n 1)
     src=$(gawk -F , "/^$name,$latestVersion,/ { print \$3 }" $csv)
     filename=$(gawk -F , "/^$name,$latestVersion,/ { print \$4 }" $csv)
-    url="${src:2}"
-    sha256=$(nix-hash --type sha256 --base32 --flat "$src")
+    url="$(dirname "${src:2}")/$filename"
```
Comment (nyanloutre) [inline maintainers/scripts/fetch-kde-qt.sh]: Maybe `$filename` is not safe here as the original name is modified

**Comment by lheckemann** [comment]
@grahamcofborg test plasma5

**Comment by nyanloutre** [comment]
@lheckemann the build seems to have timed out

**Comment by lheckemann** [comment]
Aw, that's a bummer. Since it's a minor version bump though I suppose we can still backport it to 19.03 after the branch-off.

**Comment by nyanloutre** [comment]
18.12.3 is out, I will update this PR later today

**Comment by worldofpeace** [comment]
@nyanloutre Would you mind moving this to https://kde.org/announcements/announce-applications-18.12.3.php?

I can test this locally and see if we can get a backport.

**Comment by dtzWill** [comment]
> @nyanloutre Would you mind moving this to https://kde.org/announcements/announce-applications-18.12.3.php?

Looks like this was done a while back, see the commits.  Updated title accordingly :).

**Comment by worldofpeace** [comment]
Also, do we really want this to go to staging?

**Comment by worldofpeace** [comment]
I've built all the directly affected applications and ran the NixOS test for good measure yesterday.

Minor version bumps in here are usually fine so I think this should be fine.

**Comment by Infinisil** [comment]
Yeah I think master would be better

**Comment by worldofpeace** [comment]
Merged to master in 7e8d125e19e2f82af7852f0e9f6091ca56c363da...b16b1b377bec7d4cb7430953a14b649e5b96813f

Note that https://github.com/NixOS/nixpkgs/pull/55614 and https://github.com/NixOS/nixpkgs/pull/55650 are in staging so we should anticipate a little conflict because of https://github.com/NixOS/nixpkgs/commit/4b7d9dc86800fa7b861f47ae0e6e3a8f097f9ce7

**Comment by worldofpeace** [comment]
Backport was mentioned so I will pr that to see if that's still on the table (I have no problems with it)

**Pull request closed without merging**